addon/utils/decorator-macros.js

```diff
@@ -1,70 +0,0 @@
-import { IS_EMBER_2 } from 'ember-compatibility-helpers';
-
-import { decoratorWithParams } from './decorator-wrappers';
-import extractValue from './extract-value';
-
-import { assert } from '@ember/debug';
-
-export function decoratorWithKeyReflection(fn) {
```

Comment:
I will say that I'm not 💯  on this macro system, it's been very finicky and the fact that we have so many different macros makes me feel like there's gotta be a better abstraction/system. That said, we can save it for the rewrite for v2